src/Components/LifecycleTable/lifecycleTable.tsx

```diff
@@ -0,0 +1,129 @@
+  retirement_date: 'Retirement Date',
+  systems: 'Systems'
+  };
+
+export const LifecycleTable: React.FunctionComponent<LifecycleChangesProps> = (
+  {lifecycleData}
+) => {
+
+  // Index of the currently sorted column
+  // Note: if you intend to make columns reorderable, you may instead want to use a non-numeric key
+  // as the identifier of the sorted column. See the "Compound expandable" example.
+  const [activeSortIndex, setActiveSortIndex] = React.useState<number | undefined>(undefined);
+
+  // Sort direction of the currently sorted column
+  const [activeSortDirection, setActiveSortDirection] = React.useState<'asc' | 'desc' | undefined>(undefined);
+
+  // Since OnSort specifies sorted columns by index, we need sortable values for our object by column index.
+  // This example is trivial since our data objects just contain strings, but if the data was more complex
+  // this would be a place to return simplified string or number versions of each column to sort by.
+  const getSortableRowValues = (repo: LifecycleChanges): (string | number | Date)[] => {
+    const { name, release, release_date, retirement_date, systems } = repo;
+    return [name, release, release_date, retirement_date, systems];
+  };
+
+  // Note that we perform the sort as part of the component's render logic and not in onSort.
+  // We shouldn't store the list of data in state because we don't want to have to sync that with props.
+  let sortedRepositories = lifecycleData;
+  if (activeSortIndex) {
+    sortedRepositories = lifecycleData.sort((a: LifecycleChanges, b: LifecycleChanges) => {
+      const aValue = getSortableRowValues(a)[activeSortIndex];
+      const bValue = getSortableRowValues(b)[activeSortIndex];
+      if (typeof aValue === 'number') {
+        // Numeric sort
+        if (activeSortDirection === 'asc') {
+          return (aValue as number) - (bValue as number);
+        }
+        return (bValue as number) - (aValue as number);
+      } else {
+        // String sort
+        if (activeSortDirection === 'asc') {
+          return (aValue as string).localeCompare(bValue as string);
+        }
+        return (bValue as string).localeCompare(aValue as string);
+      }
+    });
+  }
+
+  const getSortParams = (columnIndex: number): ThProps['sort'] => ({
+    sortBy: {
+      index: activeSortIndex,
+      direction: activeSortDirection,
+      defaultDirection: 'asc' // starting sort direction when first sorting a column. Defaults to 'asc'
+    },
+    onSort: (_event, index, direction) => {
+      setActiveSortIndex(index);
+      setActiveSortDirection(direction);
+    },
+    columnIndex
+  });
+
+  // In this example, we wrap all but the 1st column and make the 1st and 3rd columns sortable just to demonstrate.
+  return (
+    <Table aria-label="Sortable table" ouiaId="SortableTable">
+      <Thead>
+        <Tr>
+          <Th sort={getSortParams(0)}>{columnNames.name}</Th>
+          <Th modifier="wrap" sort={getSortParams(1)}>
```

Comment:
Trying to sort release is causing a bad bug - we should disable sort on this column if we can't handle this edge case or if we don't have good test data yet.

Reply:
What was the bug with release?

Reply:
I'm seeing `TypeError
Cannot read properties of undefined (reading 'localeCompare')`


https://github.com/user-attachments/assets/d8fd2208-755f-44c0-ac3e-82163c9b0cbd

Reply:
Oh I see, we updated the mocks for the Release field to contain "Not applicable" now. After pulling in the latest changes for digital-roadmap-backend on the develop branch it should stop giving that error